### ISSUE TITLE
Using the API to Upload reports

### ISSUE THREAD

**Issue by JWeasy** [center-for-threat-informed-defense/tram#211]
Is it possible to use the API to upload a report to TRAM?  Thanks in advance for any support.

The only API documentation I see is here:
 https://github.com/center-for-threat-informed-defense/tram/wiki/Developers#making-api-calls

**Comment by mehaase** [comment]
Not really. The API support was an after thought and doesn't expose all of the functionality in TRAM. We have helped some people build an automated submission/result system but it requires using username/password credential to login, and then scraping the HTML to extract the `csrfmiddlewaretoken` and using the HTTP headers to obtain the `sessionid` and `csrftoken` cookies. It's... not for the faint of heart.

I'll leave this issue open to keep track of this desire to see this feature added to the API. If anybody else wants to voice support for this request, please put a thumbs up on the OP.

**Comment by taketo1113** [comment]
I also wanted to use the Upload Reports API, so I created a Pull Request to add the Upload Reports API.

- #223
